module/product/src/Infrastructure/Provider/Strategy/ProductRelationAttributeValueConstraintStrategy.php

```diff
@@ -10,31 +10,52 @@
 namespace Ergonode\Product\Infrastructure\Provider\Strategy;
 
 use Ergonode\Attribute\Domain\Entity\AbstractAttribute;
-use Ergonode\Attribute\Infrastructure\Provider\AttributeValueConstraintStrategyInterface;
+use Ergonode\Attribute\Infrastructure\Provider\ContextAwareAttributeValueConstraintStrategyInterface;
+use Ergonode\EventSourcing\Infrastructure\Manager\EventStoreManagerInterface;
+use Ergonode\Product\Application\Validator\NotTheSameProduct;
+use Ergonode\Product\Domain\Entity\AbstractProduct;
+use Ergonode\SharedKernel\Domain\AggregateId;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\Constraints\Collection;
 use Ergonode\Product\Application\Validator\ProductExists;
 use Symfony\Component\Validator\Constraints\All;
 use Ergonode\Product\Domain\Entity\Attribute\ProductRelationAttribute;
 use Symfony\Component\Validator\Constraints\Uuid;
 use Symfony\Component\Validator\Constraints\NotBlank;
+use Webmozart\Assert\Assert;
 
-class ProductRelationAttributeValueConstraintStrategy implements AttributeValueConstraintStrategyInterface
+class ProductRelationAttributeValueConstraintStrategy implements ContextAwareAttributeValueConstraintStrategyInterface
 {
+    private EventStoreManagerInterface $manager;
+
+    public function __construct(EventStoreManagerInterface $manager)
+    {
+        $this->manager = $manager;
+    }
+
+
     public function supports(AbstractAttribute $attribute): bool
     {
         return $attribute instanceof ProductRelationAttribute;
     }
 
-    public function get(AbstractAttribute $attribute): Constraint
+    public function get(AbstractAttribute $attribute, ?AggregateId $aggregateId = null): Constraint
     {
+        $aggregateIdValue = null;
+        if ($aggregateId) {
+            $aggregate = $this->manager->load($aggregateId);
+            Assert::isInstanceOf($aggregate, AbstractProduct::class);
```

Comment:
Checking that this id is product id should be added as constrain "ProductExists" if i rememberer